### ISSUE TITLE
Trying to use TensorFlow

### ISSUE THREAD

**Issue by orelgueta** [cta-observatory/iact_event_types#22]
I tried using TensorFlow to see if we can improve the performance compared to the NN we have in scikit-learn.
The conclusion is, probably not. I tried various network structures, from very large networks to small ones. I also tested a couple of different loss function, learning rates, weights and a few other stuff. Did not dig too deep of course, but with none of these fairly standard TensorFlow options could I improve on the scikit-learn performance (a comparison is given in plot below). 
One interesting I found is that with TensorFlow I got somewhat better results when I did not divide the data into energy bins. Perhaps it points to the fact that we should try to optimise our energy binning also for our main model. Also, TensorFlow is much faster to train, so maybe if this becomes an issue, we can go back and try to use it.
The downside of TensorFlow is that it is more prone to predicting outliers. Very small numbers, only 0.1% of the test sample, but for some reason I almost always got a few predictions which were very far from the truth. Probably something which is easy to fix with some tuning of the network, and it doesn't affect event type classification at all, but it's something to keep in mind for the future if we go back to it.
I will leave this issue open for a little while so this study is visible if anyone is interested, but no action is required.

Plots with results are below. The first plot is the usual score vs energy, where MLP is the scikit-learn network and the TensorFlow scores are more or less the best I could get.

![image](https://user-images.githubusercontent.com/22728856/107694287-7a306280-6caf-11eb-94d1-f10900ac3f09.png)

The plot below is the distribution of the true angular error vs. the predicted one for the entire data sample (all energy values)

![image](https://user-images.githubusercontent.com/22728856/107694542-c7accf80-6caf-11eb-9c1c-a149223ffcdd.png)

This set of plots is the same as above but divided to energy bins as usual.

![image](https://user-images.githubusercontent.com/22728856/107694611-ddba9000-6caf-11eb-9c85-2e1bc8fa183d.png)
